### PR TITLE
test(flushPromiseQueue): deprecate flushPromiseQueue and give some examples

### PR DESCRIPTION
**Pull request by araujobarret**
This PR deprecates flushPromiseQueue and shows some examples how to proper handle these cases within some test scenarios.

#noChangelog

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 56f42b0f87fc965dda18257c9d277f3329305115
</p>

File: src/app/utils/tests/flushPromiseQueue.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 /**
- * test utility to wait for all promises to be resolved
+ * @deprecated consider instead:
+ *   - waiting an element to appear with `await screen.findBy`
```
Comment (damassi) [inline src/app/utils/tests/flushPromiseQueue.ts:3]: Can we add `await waitFor` to the top or second? This is generically very useful!

Comment (araujobarret) [reply src/app/utils/tests/flushPromiseQueue.ts:3]: you mean the find as one entry(1st or 2nd) and then `await waitFor(expect(mock).toHaveBeenCalledTimes(1))` for example?

Comment (damassi) [reply src/app/utils/tests/flushPromiseQueue.ts:3]: I just mean add it as a line item here:
```
 * @deprecated Consider using the following instead:
 *    - await waitFor(() => { ... })
 *    - waiting an element to appear with `await screen.findBy` 
 *    - ... 
 ```

**Review by MrSltun — APPROVED** [review]
👋👋👋

**Review by olerichter00 — APPROVED** [review]

**Review by dariakoko — APPROVED** [review]

**Pull request merged**